### PR TITLE
Adding Kafka support and upgrading to Boot 3.1.x in SMTP app.

### PR DESCRIPTION
**Pull request by gm2552**
Addressing issues #304 and #303.

### CODE REVIEW

File: spring-smtp-gateway/smtp-gateway/src/main/java/com/java/example/smtpmq/gateway/streams/SMTPMailMessageConverter.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
+/* 
+Copyright (c) 2010, NHIN Direct Project
```
Comment (trisberg) [inline spring-smtp-gateway/smtp-gateway/src/main/java/com/java/example/smtpmq/gateway/streams/SMTPMailMessageConverter.java:2]: this looks odd if this is now part of the sample source, also what is the license for this code?

File: spring-smtp-gateway/smtp-sink/src/main/java/com/java/example/smtpsink/functions/SMTPMailMessageConverter.java [hunk 2]
```diff
@@ -0,0 +1,183 @@
+/* 
+Copyright (c) 2010, NHIN Direct Project
```
Comment (trisberg) [inline spring-smtp-gateway/smtp-sink/src/main/java/com/java/example/smtpsink/functions/SMTPMailMessageConverter.java:2]: same comment here

**Review by trisberg — CHANGES_REQUESTED** [review]
I don't think we can mix in different license and copyright notices

File: spring-smtp-gateway/smtp-gateway/src/main/java/com/java/example/smtpmq/gateway/streams/SMTPMailMessageConverter.java (hunk 1 above)
Comment (gm2552) [reply spring-smtp-gateway/smtp-gateway/src/main/java/com/java/example/smtpmq/gateway/streams/SMTPMailMessageConverter.java:2]: This needs to be removed.  HNIND is not an actual entity but a place holder for something that never formed.  I'm the actual author and owner of all this code, so I need to remove these.  Thanks for catching this.

File: spring-smtp-gateway/smtp-sink/src/main/java/com/java/example/smtpsink/functions/SMTPMailMessageConverter.java (hunk 2 above)
Comment (gm2552) [reply spring-smtp-gateway/smtp-sink/src/main/java/com/java/example/smtpsink/functions/SMTPMailMessageConverter.java]: See reply on other comment.

**Comment by gm2552** [comment]
License issues have been addressed.  There should not have been any license info in these projects as they were originally generated as placeholder for an entity the never existed nor materialized.  I am the sole owner and author of this code.

**Review by trisberg — APPROVED** [review]

**Pull request merged**